### PR TITLE
Add pageable GetTags method

### PR DESCRIPTION
**Pull request by annelo-msft**
Addresses https://github.com/azure/azure-sdk-for-net/issues/19547

### CODE REVIEW

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs [hunk 1]
```diff
@@ -109,6 +110,84 @@ public async Task CanSetRepositoryProperties()
...
+
+            int pageCount = 0;
+            await foreach (var page in pages)
+            {
+                Assert.IsTrue(page.Values.Count <= pageSize);
+                pageCount++;
+            }
+
+            // Assert
+            Assert.IsTrue(pageCount >= minExpectedPages);
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs:158]: suggestion: The `Assert.That(..)` syntax is for comparisons is a bit nicer when they fail as you get a more descriptive error.

For example:
```c#
Assert.That(pageCount, Is.GreaterThanOrEqualTo(minExpectedPages));
```

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs [hunk 2]
```diff
@@ -125,6 +204,33 @@ public async Task CanGetTagProperties()
...
+            {
+                if (tag.Name.Contains("newest"))
+                {
+                    newestTagFirst = true;
+                }
+                break;
+            }
+
+            // Assert
+            Assert.IsTrue(newestTagFirst);
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs:231]: Perhaps I'm missing the intent of this test, but as long as there is a tag containing "newest" won't this always be true?

Based on the naming, I expected to see something like:

```C#
await foreach (TagProperties tag in tags)
{
    Assert.That(tag.Name.Contains("newest"));
    break;
}
```

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs:158]: There is also an Assert.GreaterOrEqual

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/TagProperties.Serialization.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Containers.ContainerRegistry
```
[the remark this answers is not in the document]
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/TagProperties.Serialization.cs:8]: We're able to[ flatten `TagAttributesBase` into `TagAttributes`](https://github.com/Azure/azure-sdk-for-js/blob/53da30b0fc693faa6d2a80f92033370194caa784/sdk/containerregistry/container-registry/swagger/containerregistry.json#L1411), but we weren't able to find a way to deserialize directly from `TagAttributesBase` to `TagAttributes` via swagger transform.  This allows us to deserialize to `TagAttributes` directly.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs (hunk 2 above)
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs:231]: I had thought with the break outside the if statement I was getting the intended logic, but yours is a lot cleaner - I'll update to use that.  Thanks!

Comment (christothes) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs:231]: Oh, sorry - I read the break as being inside the if, for some reason!

Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryClientLiveTests.cs:231]: np - I like the edit, regardless! :)

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/TagProperties.Serialization.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/TagProperties.Serialization.cs]: Added[ custom conversion](https://github.com/Azure/azure-sdk-for-net/pull/19863/files#diff-1a3d4cd8aa3b86372aa7a9e32f9700b6960131ee6da1a2be9a2440fe2ebe9637R26) to `TagList`.

**Review by christothes — APPROVED** [review]
:shipit:

**Pull request merged**